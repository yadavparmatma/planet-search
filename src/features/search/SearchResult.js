import React, {Component} from 'react';
import './search.css'

class SearchResult extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isCollapsed: false,
            collapsedPlanetName: ''
        };

        this.collapseItem = this.collapseItem.bind(this);
        this.collapseItemView = this.collapseItemView.bind(this);
    }

    collapseItem(name) {
        this.setState({isCollapsed: !this.state.isCollapsed, collapsedPlanetName: name})
    }

    collapseItemView(planet) {
        return () => {
            if (this.state.isCollapsed && planet.name === this.state.collapsedPlanetName) {
                return (
                    <div className={"planet-detail"}>
                        <hr/>
                        <span className={"name"}>{planet.name}</span>
                        <span className={"population"}>{planet.population}</span>
                        <span className={"terrain"}>{planet.terrain}</span>
                        <span className={"climate"}>{planet.climate}</span>
                    </div>
                )
            } else {
                return (
                    <div className={"population"}>
                        <hr />
                        <span className={"population"}>Planet population: {planet.population}</span>
                    </div>)
            }
        }
    }

    render() {
        return (
            <div className="search-result">
                <ul className={"planet-list"}>
                    {
                        this.props.planets.map((planet, index) => {
                            const population = planet['population'];
                            if (population === 'unknown') return null;
                            return (
                                <li key={index} className={"planet-list-item"} onClick={() => {this.collapseItem(planet.name)}}>
                                    {
                                        this.collapseItemView(planet)()
                                    }
                                </li>
                            )
                        })
                    }
                </ul>
            </div>
        );
    }
}

export default SearchResult;
