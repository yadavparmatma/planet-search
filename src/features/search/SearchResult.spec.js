import React from "react";
import renderer from 'react-test-renderer';
import SearchResult from "./SearchResult";


describe('Search', () => {
    it('should match snapshot', () => {
        const planets = [{"population":20000}, {"population":200}, {"population":190}]
        const wrapper = renderer.create(<SearchResult isLoggedIn={true} planets={planets}/>);

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
