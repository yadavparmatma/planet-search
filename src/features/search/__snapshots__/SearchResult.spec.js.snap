// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div
  className="search-result"
>
  <ul
    className="planet-list"
  >
    <li
      className="planet-list-item"
      onClick={[Function]}
    >
      <div
        className="population"
      >
        <hr />
        <span
          className="population"
        >
          Planet population: 
          20000
        </span>
      </div>
    </li>
    <li
      className="planet-list-item"
      onClick={[Function]}
    >
      <div
        className="population"
      >
        <hr />
        <span
          className="population"
        >
          Planet population: 
          200
        </span>
      </div>
    </li>
    <li
      className="planet-list-item"
      onClick={[Function]}
    >
      <div
        className="population"
      >
        <hr />
        <span
          className="population"
        >
          Planet population: 
          190
        </span>
      </div>
    </li>
  </ul>
</div>
`;
